neo4j/log/bolt_logger.go

```diff
@@ -0,0 +1,43 @@
+package log
+
+import (
+	"fmt"
+	"os"
+	"time"
+)
+
+type MessageSource int
+
+const (
+	BoltClient MessageSource = iota
+	BoltServer
+)
+
+func (src MessageSource) String() string {
+	switch src {
+	case BoltClient:
+		return "C"
+	case BoltServer:
+		return "S"
+	default:
+		return ""
+	}
+}
+
+type BoltLogger interface {
+	LogMessage(src MessageSource, context string, msg string, args ...interface{})
+}
+
+type ConsoleBoltLogger struct {
+}
+
+func (cbl *ConsoleBoltLogger) LogMessage(src MessageSource, id, msg string, args ...interface{}) {
```

Comment:
To be even more clear you now have to option to instead of having a LogMessage with a source, you can have a ServerMessage and ClientMessage function

Reply:
Great idea!